src/Sylius/Bundle/ResourceBundle/Form/DataTransformer/ResourceToIdentifierTransformer.php

```diff
@@ -48,40 +48,40 @@ public function __construct(ObjectRepository $repository, $identifier = 'id')
      */
     public function transform($value)
     {
-        if (!$value) {
-            return null;
+        if (null === $value) {
+            return '';
         }
 
-        $entity = $this->repository->findOneBy([$this->identifier => $value]);
-        if (null === $entity) {
-            throw new TransformationFailedException(sprintf(
-                'Object "%s" with identifier "%s"="%s" does not exist.',
-                $this->repository->getClassName(),
-                $this->identifier,
-                $value
-            ));
+        $class = $this->repository->getClassName();
+
+        if (!$value instanceof $class) {
+            throw new UnexpectedTypeException($value, $class);
         }
 
-        return $entity;
+        $accessor = PropertyAccess::createPropertyAccessor();
+
+        return $accessor->getValue($value, $this->identifier);
     }
 
     /**
      * {@inheritdoc}
      */
     public function reverseTransform($value)
     {
-        if (empty($value)) {
-            return '';
+        if (null === $value) {
+            return;
```

Comment:
It is unnecessary because `return;` = `return null` by default in PHP, but we have a convention to always explicitly `return null`.